File: src/main/java/onl/tesseract/entity/player/rank/StaffRank.java
package onl.tesseract.entity.player.rank;

import lombok.Getter;
import net.kyori.adventure.text.format.NamedTextColor;
import net.kyori.adventure.text.format.TextColor;
import onl.tesseract.tesseractlib.entity.Title;

@Getter
public enum StaffRank implements Rank{
    GUIDE("Guide", Title.GUIDE,NamedTextColor.YELLOW),
    ANIMATEUR("Animateur", Title.BUILDER, TextColor.fromHexString("#0088FF")),
    MEDIATEUR("Médiateur",Title.ARCHITECTE, TextColor.fromHexString("#F7DC6F")),
    MODERATEUR("Modérateur", Title.MODERATEUR,  NamedTextColor.GOLD),
    ADMINISTRATEUR("Administrateur", Title.ADMINISTRATEUR, NamedTextColor.RED),
            ;
    private final String permGroup;
    private final Title title;
    private final TextColor color;


    StaffRank(String permGroup, Title title, TextColor color) {
        this.permGroup = permGroup;
        this.title = title;
        this.color = color;
    }
}

File: src/main/java/onl/tesseract/rank/entity/StaffRank.kt
package onl.tesseract.rank.entity

import lombok.Getter
import net.kyori.adventure.text.format.NamedTextColor
import net.kyori.adventure.text.format.TextColor
import onl.tesseract.core.title.Title
import onl.tesseract.core.title.TitleService
import onl.tesseract.lib.service.ServiceContainer

private val serv = ServiceContainer[TitleService::class.java]

@Getter
enum class StaffRank(override val permGroup: String, override val title: Title, override val color: TextColor) : Rank
    {
        GUIDE("Guide", serv.getById("GUIDE"), NamedTextColor.YELLOW),
    ANIMATEUR(
        "Animateur", serv.getById("BUILDER"),
        TextColor.fromHexString("#0088FF")!!
    ),
    MEDIATEUR(
        "Médiateur", serv.getById("ARCHITECTE"),
        TextColor.fromHexString("#F7DC6F")!!
    ),
        MODERATEUR("Modérateur", serv.getById("MODERATEUR"), NamedTextColor.GOLD),
        ADMINISTRATEUR("Administrateur", serv.getById("ADMINISTRATEUR"), NamedTextColor.RED);

}
